setup.py

```diff
@@ -5,66 +5,49 @@
 
 import importlib
 from setuptools import setup, find_packages
-from sys import version_info
+from importlib.metadata import version
 from importlib import util
 
-# Interpret the version of a package depending on if python>=3.8 vs python<3.8:
-# Read: https://stackoverflow.com/questions/20180543/how-to-check-version-of-python-modules?rq=1.
-if version_info[1] <= 7:
-    import pkg_resources
+with open('requirements.txt') as f:
+    required = f.read().splitlines()
 
-    def version(s: str):
-        return pkg_resources.get_distribution(s).version
-
-else:
-    from importlib.metadata import version
+with open('requirements_test.txt') as f:
+    required_tests = f.read().splitlines()
 
 # Define extras.
 EXTRAS = {}
 EXTRAS["torch"] = (
-    ["torch==1.10.1", "torchvision==0.11.2"]
+    ["torch>=1.13.1", "torchvision>=0.15.1"]
     if not (util.find_spec("torch") and version("torch") >= "1.2")
     else []
 )
 EXTRAS["tensorflow"] = (
-    ["tensorflow==2.6.2"]
+    ["tensorflow>=2.12.0; sys_platform != 'darwin'", "tensorflow_macos>=2.12.0; sys_platform == 'darwin'"]
     if not (util.find_spec("tensorflow") and version("tensorflow") >= "2.0")
     else []
 )
 EXTRAS["captum"] = (
-    (EXTRAS["torch"] + ["captum==0.4.1"]) if not util.find_spec("captum") else []
+    (EXTRAS["torch"] + ["captum==0.6.0"]) if not util.find_spec("captum") else []
 )
 EXTRAS["tf-explain"] = (
     (EXTRAS["tensorflow"] + ["tf-explain==0.3.1"])
     if not util.find_spec("tf-explain")
     else []
 )
 EXTRAS["zennit"] = (
-    (EXTRAS["torch"] + ["zennit==0.4.5"]) if not util.find_spec("zennit") else []
-)
-EXTRAS["tutorials"] = (
-    EXTRAS["torch"] + EXTRAS["captum"] + ["pandas", "xmltodict", "tensorflow-datasets"]
+    (EXTRAS["torch"] + ["zennit==0.5.1"]) if not util.find_spec("zennit") else []
 )
-EXTRAS["tests"] = EXTRAS["captum"] + EXTRAS["tf-explain"] + EXTRAS["zennit"]
-EXTRAS["full"] = EXTRAS["tutorials"] + EXTRAS["tf-explain"] + EXTRAS["zennit"]
+EXTRAS["tests"] = required_tests
+EXTRAS["full"] = EXTRAS["captum"] + EXTRAS["tf-explain"] + EXTRAS["zennit"]
 
 # Define setup.
 setup(
     name="quantus",
-    version="0.3.5",
+    version="0.4.0",
```

Comment:
I personally would bump up the version just right before the next release, but maybe it's planned to release right after merge.

The version number is reasonable.

Reply:
Super, thanks!